nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -275,11 +280,31 @@ BtNavigator::navigateToPose()
       geometry_msgs::msg::PoseStamped goal_pose;
       blackboard_->get("goal", goal_pose);
 
-      feedback_msg->distance_remaining = nav2_util::geometry_utils::euclidean_distance(
-        feedback_msg->current_pose.pose, goal_pose.pose);
+      // Get current path points
+      nav_msgs::msg::Path current_path;
+      blackboard_->get("path", current_path);
+
+      // Calculate distance on the path
+      double distance_remaining =
+        nav2_util::geometry_utils::calculate_path_length(current_path);
+
+      if (distance_remaining <= transform_tolerance_) {
+        estimated_time_remaining_ = rclcpp::Duration::from_seconds(0.0);
+      } else {
+        // Get current speed
+        current_linear_speed_ = odom_smoother_->getTwist().linear.x;
```

Comment:
I don't think current_linear_speed_ or estimated_time_remaining_ need to be class variables - why not local to this function?

Reply:
The reason why I made these class variables is that I wanted to preserve previous values in case we don't need to update them (like when speed is less than 5mm/s). Does that make sense?

Reply:
It is computed every iteration though except when `distance_remaining > transform_tolerance_`. What you could do instead is just create those 2 variables locally and set them to `0` on instantiation. Performance wise it doesn't really matters either way, but its more of a code-cleanliness thing so that we only have class variables for things that really are used in multiple places or through various iterations.

Reply:
Yeah the more I look at this with values stored between iterations, the more I see opportunities for edge cases, please remove those class variables and initialize them as `0`s so they are `0` if whatever `if` statements don't pass to populate them.